src/components/social-media/social-media/index.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react'
+import * as S from './styles'
+
+export interface Props {
+  logo: React.FC<React.SVGProps<SVGSVGElement>>
+  url: string
+  name: string
+}
+
+const SocialMedia: React.FC<Props> = ({ logo, url, name }) => (
+  <S.Container>
+    <S.Logo>{logo}</S.Logo>
```

Comment:
Pokračování diskuse na Slacku ohledně nefunkčních ikon: 

Je zde více cest. Jedna (a asi ta nejlepší) je namapovat hodnotu tak, aby JSX kompilátor byl spokojený: 

```typescript jsx
const SocialMedia: React.FC<Props> = ({ logo, url, name }) => {
  const SocialIcon = logo
  return (
    <S.Container>
      <S.Logo>
        <SocialIcon /> 
      </S.Logo>
      <S.Link href={url}>{name}</S.Link>
    </S.Container>
  )
}
```

Každá komponenta je víceméně funkcí a JSX zápis tuto funkci zavolá. Tedy by i fungoval zápis `<S.Logo>{logo({})}</S.Logo>`. Některé komponenty tento případ řešili přes `renderLogo` namísto `logo`, kde bylo jasné, že se jedná o funkci při použití komponenty, ale i v její implementaci. Nicméně toto řešení vyžaduje zanoření do funkce `renderIcon: () => SlackIcon` u každého použití.